crowbar_framework/app/models/service_object.rb

```diff
@@ -1203,31 +1203,15 @@ def apply_role(role, inst, in_queue, bootstrap = false)
         node_attr_cache[node_name]["windows"] || node_attr_cache[node_name]["admin"]
       end
 
-      begin
-        apply_locks = []
-        # do not use a map to set apply_locks: if there's a failure, we need the
-        # variable to contain the locks that were acquired; with a map, the
-        # variable would be empty.
-        nodes_to_lock.each do |node|
-          apply_lock = Crowbar::Lock::SharedNonBlocking.new(
-            logger: Rails.logger,
-            path: "/var/chef/cache/pause-file.lock",
-            node: node,
-            owner: "apply_role-#{role.name}-#{inst}-#{Process.pid}",
-            reason: "apply_role(#{role.name}, #{inst}, #{in_queue}) pid #{Process.pid}"
-          ).acquire
-          apply_locks.push(apply_lock)
-        end
-      rescue Crowbar::Error::LockingFailure => e
-        Rails.logger.progress("Failed to apply role #{role.name}")
-        message = "Failed to apply the proposal: #{e.message}"
+      owner = "apply_role-#{role.name}-#{inst}-#{Process.pid}"
+      reason = "apply_role(#{role.name}, #{inst}, #{in_queue}) pid #{Process.pid}"
+      apply_locks, errors = lock_and_wait_for_chef_clients(nodes_to_lock, owner, reason)
```

Comment:
I think that the var name was kept for easy diff between the old changes and new, I dont mind changing it :)